crates/iroha_kagami/src/genesis.rs

```diff
@@ -1,23 +1,183 @@
-use std::io::{BufWriter, Write};
+use std::{
+    io::{BufWriter, Write},
+    path::PathBuf,
+};
 
 use clap::Subcommand;
+use color_eyre::eyre::WrapErr as _;
+use iroha_data_model::{isi::InstructionBox, parameter::Parameters, prelude::*};
+use iroha_executor_data_model::permission::{
+    domain::CanRegisterDomain, parameter::CanSetParameters,
+};
+use iroha_genesis::{GenesisBuilder, GenesisSpec, GENESIS_ACCOUNT_ID};
+use iroha_test_samples::{gen_account_in, ALICE_ID, BOB_ID, CARPENTER_ID};
 
 use crate::{Outcome, RunArgs};
 
-mod generate;
-mod sign;
+/// Generate a genesis configuration and standard-output in JSON format
+#[derive(clap::Args, Debug, Clone)]
+pub struct Args {
+    /// Relative path from the directory of output file to the executor.wasm file
+    #[clap(long, value_name = "PATH")]
+    executor: PathBuf,
+    /// Relative path from the directory of output file to the directory that contains *.wasm libraries
+    #[clap(long, value_name = "PATH")]
+    wasm_dir: PathBuf,
+    #[clap(subcommand)]
+    mode: Option<Mode>,
+}
 
-#[derive(Debug, Clone, Subcommand)]
-pub enum Args {
-    Sign(sign::Args),
-    Generate(generate::Args),
+#[derive(Subcommand, Debug, Clone, Default)]
+pub enum Mode {
+    /// Generate default genesis
+    #[default]
+    Default,
+    /// Generate synthetic genesis with the specified number of domains, accounts and assets.
+    ///
+    /// Synthetic mode is useful when we need a semi-realistic genesis for stress-testing
+    /// Iroha's startup times as well as being able to just start an Iroha network and have
+    /// instructions that represent a typical blockchain after migration.
+    Synthetic {
+        /// Number of domains in synthetic genesis.
+        #[clap(long, default_value_t)]
+        domains: u64,
+        /// Number of accounts per domains in synthetic genesis.
+        /// The total number of accounts would be `domains * assets_per_domain`.
+        #[clap(long, default_value_t)]
+        accounts_per_domain: u64,
+        /// Number of assets per domains in synthetic genesis.
+        /// The total number of assets would be `domains * assets_per_domain`.
+        #[clap(long, default_value_t)]
+        assets_per_domain: u64,
+    },
 }
 
 impl<T: Write> RunArgs<T> for Args {
     fn run(self, writer: &mut BufWriter<T>) -> Outcome {
-        match self {
-            Args::Sign(args) => args.run(writer),
-            Args::Generate(args) => args.run(writer),
+        let Self {
+            executor,
+            wasm_dir,
+            mode,
+        } = self;
+
+        let chain = ChainId::from("00000000-0000-0000-0000-000000000000");
+        let builder = GenesisBuilder::new(chain, executor, wasm_dir);
```

Comment:
IMO here Kagami should provide block creation time as the current timestamp, or a timestamp specified in the args (helpful for scripts and swarm).